tests/scripts/thread-cert/Cert_5_5_02_LeaderReboot.py

```diff
@@ -89,6 +95,102 @@ def test(self):
+
+        _rpkts.filter_mle_cmd(consts.MLE_ADVERTISEMENT).must_next().must_verify(
+            lambda p: {0, 11, 9} == set(p.mle.tlv.type))
+
+        # Step4: Router_1 MUST attempt to reattach to its original partition by
+        # sending MLE Parent Requests to the All-Routers multicast
+        # address (FFxx::xx) with a hop limit of 255.
+        _rpkts.filter_mle_cmd(
+            consts.MLE_PARENT_REQUEST).must_next().must_verify(
+                lambda p: {1, 3, 14, 18} == set(p.mle.tlv.type))
+        lreset_start = _rpkts.index
+
+        # Step 5: Leader MUST NOT respond to the MLE Parent Requests
+        _lpkts.filter_mle_cmd(consts.MLE_PARENT_RESPONSE).must_not_next()
+
+        # Step 6:Router_1 MUST attempt to attach to any other Partition
+        # within range by sending a MLE Parent Request.
+        _rpkts.filter_mle_cmd(
+            consts.MLE_PARENT_REQUEST).must_next().must_verify(
+                lambda p: {1, 3, 14, 18} == set(p.mle.tlv.type))
+        lreset_stop = _rpkts.index
+
+        # Step3: The Leader MUST stop sending MLE advertisements.
+        leader_pkts.range(lreset_start, lreset_stop).filter_mle_cmd(
+            consts.MLE_ADVERTISEMENT).must_not_next()
+
+        # Step 7: Take over leader role of a new Partition and
+        # begin transmitting MLE Advertisements
+        _temppkts = _rpkts.copy()
+        _temppkts.filter_mle_cmd(
+            consts.MLE_ADVERTISEMENT).must_next().must_verify(
+                lambda p: {0, 11, 9} == set(p.mle.tlv.type))
+
+        # Step 8: Router_1 MUST respond with an MLE Child Update Response,
+        # with the updated TLVs of the new partition
+        _rpkts.filter_mle_cmd(14).must_next().must_verify(
+            lambda p: {0, 1, 11, 19} < set(p.mle.tlv.type))
+
+        # Step 9: The Leader MUST send properly formatted MLE Parent
+        # Requests to the All-Routers multicast address
+        _lpkts.filter_mle_cmd(
+            consts.MLE_PARENT_REQUEST).must_next().must_verify(
+                lambda p: {1, 3, 14, 18} == set(p.mle.tlv.type))
+
+        # Step 10: Router_1 MUST send an MLE Parent Response
+        _rpkts.filter_mle_cmd(
+            consts.MLE_PARENT_RESPONSE).must_next().must_verify(
+                lambda p: {0, 11, 5, 4, 3, 16, 15, 18} < set(p.mle.tlv.type))
+
+        # Step 11: Leader send MLE Child ID Request
+        _lpkts.filter_mle_cmd(consts.MLE_CHILD_ID_REQUEST).must_next(
+        ).must_verify(
+            lambda p: {4, 5, 1, 2, 18, 13, 10, 12, 9, 22} < set(p.mle.tlv.type))
+
+        #Step 12: Router_1 send MLE Child ID Response
+        _rpkts.filter_mle_cmd(
+            consts.MLE_CHILD_ID_RESPONSE).must_next().must_verify(
+                lambda p: {0, 11, 10, 12, 9} < set(p.mle.tlv.type))
+
+        #Step 13: Leader send an Address Solicit Request
+        _lpkts.filter_coap_request("/a/as").must_next().must_verify(
+            lambda p: p.coap.tlv.ext_mac_addr and p.coap.tlv.rloc16 and p.coap.
```

Comment:
**Note that 0x0000 is actually a valid Leader RLOC16**. In such case, this verification condition would fail. 
`pktverify` suggests another way of checking if the field does not exist:

```suggestion
            lambda p: p.coap.tlv.ext_mac_addr and p.coap.tlv.rloc16 is not nullField and p.coap.
```

BTW, better use `p.coap.tlv.status != 0` to check for non-zero `p.coap.tlv.status`.